### PR TITLE
Make DNS not need its own token

### PR DESCRIPTION
**Pull request by thockin**
Different than the previous attempt, this still uses the DNS token, but enables kube2sky to work in insecure mode without it.

### CODE REVIEW

**Comment by thockin** [comment]
@mikedanese @lavalamp This still doesn't quite work properly  If I troigger the fallback mode, I get:

```
/ # /kube2sky -v=4 -domain=cluster.local 
I0620 03:56:44.311909      25 kube2sky.go:389] Etcd server found: http://127.0.0.1:4001
I0620 03:56:45.313408      25 kube2sky.go:455] Using https://10.0.0.1:443 for kubernetes master
I0620 03:56:45.313674      25 kube2sky.go:456] Using kubernetes API v1beta3
E0620 03:56:45.315571      25 reflector.go:136] Failed to list *api.Service: Get https://10.0.0.1:443/api/v1beta3/services: EOF
E0620 03:56:45.315972      25 reflector.go:136] Failed to list *api.Endpoints: Get https://10.0.0.1:443/api/v1beta3/endpoints: EOF
E0620 03:56:46.316466      25 reflector.go:136] Failed to list *api.Service: Get https://10.0.0.1:443/api/v1beta3/services: EOF
E0620 03:56:46.317129      25 reflector.go:136] Failed to list *api.Endpoints: Get https://10.0.0.1:443/api/v1beta3/endpoints: EOF
E0620 03:56:47.317894      25 reflector.go:136] Failed to list *api.Service: Get https://10.0.0.1:443/api/v1beta3/services: EOF
E0620 03:56:47.318669      25 reflector.go:136] Failed to list *api.Endpoints: Get https://10.0.0.1:443/api/v1beta3/endpoints: EOF
```

It chose v1beta3, I guess, and it found the master.  But it's not working.

**Comment by thockin** [comment]
@vishh @ArtfulCoder also --kube_master_url is not working:

```
# /kube2sky -v=4 -domain=cluster.local -kube_master_url=http://10.0.0.1:443
I0620 04:14:24.737157      48 kube2sky.go:389] Etcd server found: http://127.0.0.1:4001
I0620 04:14:25.738255      48 kube2sky.go:455] Using http://10.0.0.1:443 for kubernetes master
I0620 04:14:25.738578      48 kube2sky.go:456] Using kubernetes API v1
E0620 04:14:25.839975      48 reflector.go:136] Failed to list *api.Endpoints: Get http://10.0.0.1:443/api/v1/endpoints: net/http: transport closed before response was received
E0620 04:14:25.840397      48 reflector.go:136] Failed to list *api.Service: Get http://10.0.0.1:443/api/v1/services: net/http: transport closed before response was received
E0620 04:14:26.941659      48 reflector.go:136] Failed to list *api.Endpoints: Get http://10.0.0.1:443/api/v1/endpoints: net/http: transport closed before response was received
E0620 04:14:26.942090      48 reflector.go:136] Failed to list *api.Service: Get http://10.0.0.1:443/api/v1/services: net/http: transport closed before response was received
```

File: cluster/addons/dns/kube2sky/Changelog [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 
 
-## Version 1.7 (May 25 2015 Vishnu Kannan <vishnuk@google.com>)
-- Adding support for headless services. All pods backing a headless service is addressible via DNS RR.
+## Version 1.9 (May 28 2015 Abhishek Shah <abshah@google.com>)
+- Add SRV support.
 
 
-## Version 1.8 (May 18 2015 Vishnu Kannan <vishnuk@google.com>)
+## Version 1.8 (May 28 2015 Vishnu Kannan <vishnuk@google.com>)
```
Comment (roberthbailey) [inline cluster/addons/dns/kube2sky/Changelog]: This version's PR was sent on may 22nd and merged on june 2nd.

File: cluster/addons/dns/kube2sky/Changelog [hunk 2]
```diff
@@ -1,12 +1,20 @@
-* Fri May 15 2015  Tim Hockin <thockin@google.com>
-  - First Changelog entry
-  - Current version is 1.4
+## Version 1.10 (Jun 19 2015 Tim Hockin <thockin@google.com>)
+- Fall back on service account tokens if no other auth is specified.
 
 
-## Version 1.7 (May 25 2015 Vishnu Kannan <vishnuk@google.com>)
-- Adding support for headless services. All pods backing a headless service is addressible via DNS RR.
+## Version 1.9 (May 28 2015 Abhishek Shah <abshah@google.com>)
```
Comment (roberthbailey) [inline cluster/addons/dns/kube2sky/Changelog]: This version's PR was sent on June 1st and merged on June 5th.

Comment (thockin) [inline cluster/addons/dns/kube2sky/Changelog]: I got the date that was missing from the commits or something.  The exact
dates are not so important.

**Comment by thockin** [comment]
@mikedanese So basically your fix works, everything else now fails.

**Comment by mikedanese** [comment]
I will look into what's going on here.

**Comment by mikedanese** [comment]
I checked out your branch and started kube2sky with
```
"args": ["-domain=cluster.local", "-v=10"],
```
It appears to have fallenback to inClusterConfig correctly. Here are the logs: https://gist.github.com/mikedanese/f72df13b04e49e128e97

**Comment by thockin** [comment]
So what part is not working? I am out now,but I will try to burn the
cluster down and recreate to verify

**Comment by thockin** [comment]
after burn-down it seems to work.  I'll update this PR later

**Comment by mikedanese** [comment]
If you wait for #10264 to merge and rebase, even if you don't pass a kubeconfig the connection to the apiserver will still be secured.

**Comment by thockin** [comment]
ETA?

**Comment by thockin** [comment]
OK, this PR is retooled.

* DNS no longer needs a special token!  @pires and countless others for a "hip, hip, hooray!".  NB: this depends on master-side code checked in TODAY, so it won't just work on random platforms without an update.  Big thanks to @lavalamp and @mikedanese 

* New kube2sky 1.10 release

* All using v1 API

I tested it manually and also in a non-default namespace.  We'll watch here for e2e.

Ready for review, but DO NOT MERGE until I push the actual 1.10 container image.


@ArtfulCoder @lavalamp @dchen1107  for LGTMs

**Comment by pires** [comment]
@thockin thanks for the heads-up. hip, hip hooray! :clap:

**Comment by thockin** [comment]
@ixdy How do I make k8s-bot retest?

**Comment by ixdy** [comment]
retest this please

**Comment by thockin** [comment]
Can you help me interpret this?  I don't see why it is failing.

**Comment by ixdy** [comment]
It seems like the DNS pod is stuck pending?
```
[15] INFO: 5 / 6 pods in namespace 'default' are running and ready (600 seconds elapsed)
[15] INFO: POD                NODE                    PHASE    CONDITIONS
[15] INFO: kube-dns-v4-cqzws  pull-e2e-3-minion-c56f  Pending  [{Ready False}]
```

**Comment by ixdy** [comment]
The previous runs here all failed in the same way.

**Comment by thockin** [comment]
Oh, duh, of course.  I have not pushed the tagged container yet because I
don't want people using it.  chicken-egg.  will think on it tomorrow.

**Comment by ArtfulCoder** [comment]
LGTM
existing e2e dns test should be enought to cover this (though we probably dont get coverage on all combinations of apiserver url specification options)

**Comment by ArtfulCoder** [comment]
Oncall Note: that the container image needs to be pushed out before this PR can be merged. If e2e for PR run is green, it would probably mean that the container image is pushed out.

**Comment by ArtfulCoder** [comment]
@thockin might want to squash commits into 1 ?

**Comment by roberthbailey** [comment]
@dchen1107 @quinton-hoole (need a couple of leads to LGTM -- suggest others if you are overloaded).

**Comment by quinton-hoole** [comment]
@k8s-bot Test this please.

**Comment by thockin** [comment]
I like the commits.  Keeping.  I pushed the image and tests will pass, I expect.

**Comment by quinton-hoole** [comment]
Agreed, Keeping the commits separate seems like a good idea.  Most of this is fairly innocuous config and version number handling stuff.  Looks pretty safe to me, and easy to roll back if required. LGTM.

ok-to-merge as soon as Jenkins GCE e2e passes.

**Comment by ixdy** [comment]
@k8s-bot test this please

I think case may matter...

**Comment by roberthbailey** [comment]
LGTM as well.

**Pull request merged**